aodntools/timeseries_products/gridded_timeseries.py

```diff
@@ -0,0 +1,262 @@
+    :param max_separation: max separation allowed for instruments
+    :param depth_bins_increment: in case no depth bins provided this is the increment for the calculated bins
+    :param base_path: path where the result file will be written
+    :return: interpolated dataset
+    """
+
+    with xr.open_dataset(file_name) as nc:
+        site_code = nc.site_code
+        ## get global attributes
+        global_attributes = nc.attrs
+
+        ## get lat/lon
+        longitude_mean = nc.LONGITUDE.mean()
+        latitude_mean = nc.LATITUDE.mean()
+
+        ## in case no depth bins provided, create depth bins to the nearest rounded depth
+        ## if provided, adjust to the min-max registered depth
+        if not depth_bins:
+            depth_bins = make_depth_bins(nc=nc, increment=depth_bins_increment)
+        else:
+            depth_bins = [float(depth) for depth in depth_bins]     # in case depth_bins provided through inline arguments
+            depth_bins = adjust_depth_bins(depth_bins, nc.DEPTH.min(), nc.DEPTH.max())
+
+        ## create empty containers
+        time_0 = pd.to_datetime('1950-01-01T00:00:00')
+        time_min = nc.TIME.values.min()
+        depth_bin_len = len(depth_bins)
+        VoI_binned = xr.DataArray(np.full((depth_bin_len, 1), np.nan), coords=[[time_0], depth_bins],
+                                  dims=['TIME', 'DEPTH'])
+        VoI_ndepths = xr.DataArray(np.full(1, 0), coords=[[time_0]], dims=['TIME'])
+
+        ## group nc by individual timestamps
+        VoI_grouped = nc.groupby('TIME')
+
+
+        for timestamp, (name, group) in enumerate(VoI_grouped):
+            time = [name]
+            n_depths = int(len(group[VoI]))
+
+            if n_depths >= 2:
+                VoI_values = list(group[VoI].values)
+                depth = list(group.DEPTH.values)
+                ## sort depths
+                depth, VoI_values = sort_depths(depth, VoI_values)
+
+                ## check for max separation
+                depth_mask = get_depth_mask(depth_bins=depth_bins, depths=depth, max_separation=max_separation)
+                ## do the interpolation
+                VoI_gridded = np.interp(depth_bins, depth, VoI_values, left=np.nan, right=np.nan)
+                ## set masked depth bins to zero
+                VoI_gridded = VoI_gridded * depth_mask
+                VoI_gridded[VoI_gridded == 0] = np.nan
+
+            else:
+                VoI_gridded = np.full((depth_bin_len, 1), np.nan)
+
+            VoI_temp = xr.DataArray(VoI_gridded.reshape(depth_bin_len, 1), coords=[time, depth_bins],
+                                    dims=['TIME', 'DEPTH'])
+            VoI_ndepths_temp = xr.DataArray([n_depths], coords=[time], dims=['TIME'])
+
+            VoI_binned = xr.concat([VoI_binned, VoI_temp], dim='TIME')
+            VoI_ndepths = xr.concat([VoI_ndepths, VoI_ndepths_temp], dim='TIME')
+
+    VoI_interpolated = xr.Dataset({VoI: VoI_binned.astype(float),
```

Comment:
do the conversion to float earlier so all operations are faster.

Reply:
:+1: I agree.
Also, this conversion doesn't seem to have any effect on the final output file, which has TEMP as a `double`. I think a `float` would be sufficient.